Added support for ACLs on port groups.

anfredette opened the pull request:
Added support for ACLs on port groups.
 - Using ovn-nbctl acl-add/del as a model, an entityType parameter was added to
the API which could be either PORT_GROUP or LOICAL_SWITCH.
- In order to maintain backward compatibility of the existing APIs, a new set
of APIs called ACLAddEntity/ACLDelEntity/ACLListEntity were added.
- Note: both versions of the APIs call the same underlying
implementation functions.
    
Also made a minor optimization suggested by vtolstov on pr-117.
    
Signed-off-by: Andre Fredette <afredette@redhat.com>

hzhou8 commented:
Thanks @anfredette . LGTM!

A minor comment is that it would be better to split the fixes such as for the problem when external_ids == nil in separate commits. Let me know if you'd like to do that.

anfredette commented:
I agree. I’ll split it out.

Thanks,
Andre

On Sun, Jan 3, 2021 at 5:21 PM Han Zhou <notifications@github.com> wrote:

> Thanks @anfredette <https://github.com/anfredette> . LGTM!
>
> A minor comment is that it would be better to split the fixes such as for
> the problem when external_ids == nil in separate commits. Let me know if
> you'd like to do that.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/eBay/go-ovn/pull/122#issuecomment-753691164>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAHY7X3QKMLWM2U3JRHXDDTSYD3Y3ANCNFSM4VGJWSRA>
> .
>

anfredette commented:
I've updated this PR as follows:
1. Created Issue #124 and split out fix as PR  #125.
2. Rebased this PR on PR #124.
3. Optimized getACLUUIDByRow() 
   - when we have the row UUID, we don't have to loop through the table to find it.

softwarejl reviewed (approved):
LGTM

hzhou8 commented on acl.go:
```diff
@@ -34,120 +34,160 @@ type ACL struct {
+		condition = libovsdb.NewCondition("name", "==", entity)
+		table = TableLogicalSwitch
+	case PORT_GROUP:
+		condition = libovsdb.NewCondition("_uuid", "==", stringToGoUUID(entity))
```
Why using UUID for port_group (while we use "name" for logical_switch). Is it better to use "name" so that it is consistent between the two entity types?

anfredette replied:
Done.

anfredette commented:
Per Han's request above, and discussion on Slack, I've changed the port group ACL APIs to use the PG name instead of the PG UUID.

NOTE: I've kept this change as a separate commit to facilitate reviews.  After I get the thumbs up, I will squash the last two commits.

hzhou8 reviewed (changes requested):
Sorry that I have to put this on hold again because of the external-ids commit. Please see my comment in https://github.com/eBay/go-ovn/pull/125 for the same commit.

anfredette commented:
This PR has been reworked on top of PR-125.
